### PR TITLE
By default digitize single partition ME0 geometry with custom parameters

### PR DESCRIPTION
**Pull request by nickmccoll**
PR #18113 Updated the digitizer to use either the built-in geometry or the custom one defined in the parameter set. For single partition geometries, which will not be used in the Muon TDR, this resulted in a single partition digitization. In order to allow for better default behavior in geometries such as D4 the default is to now use the custom geometry whenever there is a single partition geometry.

The "useBuiltInGeo" parameter wasn't useful for geometries other than single partition ones. It would return an error if you tried to use it on a multi-partition one. Instead it was replaced with "useCusGeoFor1PartGeo" which is only used on single partition geometries. No change in the reconstruction for the default, D12 geometry. 

D4 geometry (1 partition) before this PR:
[hitMap_D4_beforePR.pdf](https://github.com/cms-sw/cmssw/files/911793/hitMap_D4_beforePR.pdf)

D4 geometry (1 partition) after this PR:
[hitMap_D4_afterPR.pdf](https://github.com/cms-sw/cmssw/files/911795/hitMap_D4_afterPR.pdf)

D12 geometry (8 partition) before this PR:
[hitMap_D12_beforePR.pdf](https://github.com/cms-sw/cmssw/files/911796/hitMap_D12_beforePR.pdf)

D12 geometry (8 partition) after this PR:
[hitMap_D12_afterPR.pdf](https://github.com/cms-sw/cmssw/files/911798/hitMap_D12_afterPR.pdf)

There is a displacement in local (chamber) coordinates between D4 and D12. This is due tot he placement of the eta partitions within the D4 geo. It doesn't affect segment reconstruction as it is simply a uniform displacement in Y.

### CODE REVIEW

**Comment by civanch** [comment]
please test

File: SimMuon/GEMDigitizer/src/ME0ReDigiProducer.cc [hunk 1]
```diff
@@ -183,11 +184,20 @@ void ME0ReDigiProducer::beginRun(const edm::Run&, const edm::EventSetup& eventSe
 	eventSetup.get<MuonGeometryRecord>().get(hGeom);
 	geometry= &*hGeom;
 
+	const auto& chambers = geometry->chambers();
+	if(!chambers.size())
```
Comment (nickmccoll) [inline SimMuon/GEMDigitizer/src/ME0ReDigiProducer.cc]: Done!

**Comment by kpedro88** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
+1

**Comment by kpedro88** [comment]
+1
Comparison changes only in ME0 digis and rechits for 2023 workflows, as expected.

**Comment by davidlange6** [comment]
+1

**Pull request merged**